packages/client/jest.config.js

```diff
@@ -1,5 +1,11 @@
 module.exports = {
   preset: 'ts-jest',
+  globals: {
+    'ts-jest': {
+      tsconfig: 'tsconfig.json',
+      isolatedModules: true,
```

Comment:
The reason behind a few hours being wasted... (ts-jest hanging in CI) Surprisingly it worked fine for a long time, until I made some small type changes.

Reply:
Weird, shouldn't that already be set? `tsconfig.json` inherits the config from the root repo that already enables `isolatedModules`.

Reply:
`isolatedModules` is misleading because what it does is to disable type checking.
https://kulshekhar.github.io/ts-jest/docs/getting-started/options/#options
https://github.com/kulshekhar/ts-jest/issues/3507

Side note:
Btw, since the codebase was migrated to use `@swc/jest`, we also lost all test type checking.
(I'll add this to the tech debt list so that we can fix this asap)